readthedocs/core/views/hooks.py

```diff
@@ -75,9 +92,17 @@ def build_branches(project, branch_list):
         not_building - a list of branches that we won't build
     """
     for branch in branch_list:
-        versions = project.versions_from_branch_name(branch)
+        commit = branch['last_commit']
+        versions = project.versions_from_branch_name(branch['name'])
         to_build = set()
         not_building = set()
+        if _contains_skip_mark(commit):
+            log.info('(Branch Build) Skip build %s', project.slug)
```

Comment:
Add the reason of the skip, like "because contains skip mark" or something _better_ :)